File: exercises/python/shared_observability/baggage_helpers.py
"""Helpers for reading and writing OpenTelemetry Baggage."""

from __future__ import annotations

from opentelemetry import baggage, context


def set_baggage(key: str, value: str) -> context.Context:
    """Attach a key/value pair to the current OTel baggage.

    Returns the new context (callers can pass it forward or attach it).
    """
    ctx = baggage.set_baggage(key, value)
    context.attach(ctx)
    return ctx


def get_baggage(key: str) -> str | None:
    """Read a value from the current OTel baggage, or None if absent."""
    return baggage.get_baggage(key)

File: exercises/dotnet/SharedObservability/BaggageHelpers.cs
using System.Diagnostics;
using OpenTelemetry;

namespace SharedObservability;

/// <summary>
/// Static helpers for reading and writing OpenTelemetry Baggage entries.
/// Baggage propagates domain identifiers (order ID, customer ID, etc.) across
/// service boundaries so that downstream spans and logs carry business context.
/// </summary>
public static class BaggageHelpers
{
    /// <summary>
    /// Retrieves a baggage value by key, returning null if not present.
    /// </summary>
    public static string? Get(string key)
    {
        return Baggage.GetBaggage(key);
    }

    /// <summary>
    /// Sets a baggage key-value pair on the current activity context.
    /// </summary>
    public static void Set(string key, string value)
    {
        Baggage.SetBaggage(key, value);
    }

    /// <summary>
    /// Removes a baggage entry by key.
    /// </summary>
    public static void Remove(string key)
    {
        Baggage.RemoveBaggage(key);
    }

    /// <summary>
    /// Returns all current baggage entries as key-value pairs.
    /// </summary>
    public static IReadOnlyDictionary<string, string> GetAll()
    {
        return Baggage.GetBaggage()
            .ToDictionary(kvp => kvp.Key, kvp => kvp.Value);
    }
}
